### PR TITLE
neovim: fix ruby provider warning

### PR DESCRIPTION
**Pull request by BirdeeHub**
Fixing this warning from checkhealth

```
provider.ruby: require("provider.ruby.health").check()

Ruby provider (optional) ~
- Ruby: ruby 3.3.5 (2024-09-03 revision ef084cc8f4) [x86_64-linux]
- Host: /nix/store/l354hgiahcjd81qazlc870104i26lm9b-neovim-0.10.2birdeevim/bin/birdeeVim-ruby
- OK Latest "neovim" gem is installed: /home/birdee/.local/share/gem/ruby/3.3.0/gems/neovim-0.10.0/lib/neovim/connection.rb:1: warning: logger was loaded from the standard library, but will no longer be part of the default gems starting from Ruby 3.5.0. You can add logger to your Gemfile or gemspec to silence this warning. 0.10.0
```

steps taken:

added `gem 'logger'` to Gemfile

ran the following on the directory

nix run --no-write-lock-file github:BirdeeHub/neovim_ruby_updater

https://github.com/BirdeeHub/neovim_ruby_updater

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin

Also currently using the same thing in my personal configuration [seen here](https://github.com/BirdeeHub/birdeeSystems/tree/main/common/birdeevim/overlays/ruby_provider) and it works

### CODE REVIEW

File: pkgs/applications/editors/neovim/ruby_provider/gemset.nix [hunk 1]
```diff
@@ -1,13 +1,23 @@
...
     groups = ["default"];
     platforms = [];
     source = {
       remotes = ["https://rubygems.org"];
-      sha256 = "1a5adcb7bwan09mqhj3wi9ib52hmdzmqg7q08pggn3adibyn5asr";
+      sha256 = "1a5fsgchkpcca4wf3pipbb2jbj523l7fbaq37j10cr0yymwlkc7z";
       type = "gem";
     };
-    version = "1.7.2";
+    version = "1.7.5";
```
Comment (mrcjkb) [inline pkgs/applications/editors/neovim/ruby_provider/gemset.nix:20]: question: Is this update required for the logger? It looks unrelated.

Comment (BirdeeHub) [reply pkgs/applications/editors/neovim/ruby_provider/gemset.nix:20]: it is but it works? I can undo that part I guess but this is what bundix did

All I did was add `gem 'logger'` to the gemfile and then run this.

```nix
        default = pkgs.writeShellScriptBin "nvim_ruby_updater" ''
          echo "Running 'bundle update'..."
          ${rubyEnv}/bin/bundle update
          echo "Running 'bundix'..."
          ${pkgs.bundix}/bin/bundix
        '';
```

I would assume that the best msgpack version to use would be the one that bundix says but idk

Thats a generated file

Comment (BirdeeHub) [reply pkgs/applications/editors/neovim/ruby_provider/gemset.nix:20]: `nix run --no-write-lock-file github:BirdeeHub/neovim_ruby_updater`

I made a little util for updating it

Comment (BirdeeHub) [reply pkgs/applications/editors/neovim/ruby_provider/gemset.nix:20]: Im sure I could do that though but it doesnt seem to matter

Comment (khaneliman) [reply pkgs/applications/editors/neovim/ruby_provider/gemset.nix:20]: I don't think you need to rip it out, it was more about organizing commits into logical units to organize related changes. One was adding a dependency and one was a version bump of existing dependencies.

Comment (BirdeeHub) [reply pkgs/applications/editors/neovim/ruby_provider/gemset.nix:20]: Should they be separate PR or separate commits? I can do separate commits easy enough I guess?

Comment (khaneliman) [reply pkgs/applications/editors/neovim/ruby_provider/gemset.nix:20]: > Should they be separate PR or separate commits? I can do separate commits easy enough I guess?

Just separate commits to indicate one is adding a dependency to silence warnings and another to bump the version to latest release.

Comment (BirdeeHub) [reply pkgs/applications/editors/neovim/ruby_provider/gemset.nix:20]: honestly I dont know what bundled with does either

I kinda would rather just leave it as one because I know it works and I dont wanna break it

Comment (mrcjkb) [reply pkgs/applications/editors/neovim/ruby_provider/gemset.nix:20]: I think if using `gem 'logger'` bumps another dependency, then it's fine to be in a single commit.

Comment (khaneliman) [reply pkgs/applications/editors/neovim/ruby_provider/gemset.nix:20]: > I dont know what happens if I manually downgrade it without changing bundled with is the issue XD

Bundled with looks like just metadata for explaining the version the command was run with.

Comment (SuperSandro2000) [reply pkgs/applications/editors/neovim/ruby_provider/gemset.nix:20]: Does this really matter? I don't think it does much. Bundler bumped the package while regenerating the lock file which should be totally fine here.

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/applications/editors/neovim/ruby_provider/gemset.nix (hunk 1 above)
Comment (khaneliman) [reply pkgs/applications/editors/neovim/ruby_provider/gemset.nix:20]: Never said that mattered, I don't know why it was hung up on.. lol

**Comment by khaneliman** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 357902`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 9 packages built:</summary>
  <ul>
    <li>gnvim</li>
    <li>lunarvim</li>
    <li>neovide</li>
    <li>neovim</li>
    <li>neovim-qt</li>
    <li>neovim-remote</li>
    <li>neovim-remote.dist</li>
    <li>nvimpager</li>
    <li>vim-startuptime</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 9 packages built:</summary>
  <ul>
    <li>gnvim</li>
    <li>lunarvim</li>
    <li>neovide</li>
    <li>neovim</li>
    <li>neovim-qt</li>
    <li>neovim-remote</li>
    <li>neovim-remote.dist</li>
    <li>nvimpager</li>
    <li>vim-startuptime</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 7 packages built:</summary>
  <ul>
    <li>gnvim</li>
    <li>lunarvim</li>
    <li>neovide</li>
    <li>neovim</li>
    <li>neovim-qt</li>
    <li>nvimpager</li>
    <li>vim-startuptime</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 7 packages built:</summary>
  <ul>
    <li>gnvim</li>
    <li>lunarvim</li>
    <li>neovide</li>
    <li>neovim</li>
    <li>neovim-qt</li>
    <li>nvimpager</li>
    <li>vim-startuptime</li>
  </ul>
</details>

File: pkgs/applications/editors/neovim/ruby_provider/gemset.nix (hunk 1 above)
Comment (BirdeeHub) [reply pkgs/applications/editors/neovim/ruby_provider/gemset.nix:20]: Yeah it didnt seem important and it worked.

I figure because its a newer version of ruby in nixpkgs now the current version is best

**Review by mrcjkb — APPROVED** [review]
